Utils/Dataflow/069_upload2es/load_data.sh

```diff
@@ -12,6 +12,38 @@ PARAMETERS:
 "
 }
 
+while [[ $# > 1 ]]
+do
+  key="$2"
```

Comment:
This looks a bit strange, aspecially because you use command `shift` later in the loop.

**About the general logic**

I guess you wanted to leave input filename in the first argument, but the thing is that FILENAME is a) an optional argument (which is reflected in usage massage via square brackets around the argument name "FILENAME") and b) there might be more than one file, which name should be passed to the script (which is reflected by dots after the argument name).
It means, that FILENAME simply can not go as the first argument -- only as the last (or else we will have problems locating and parsing other arguments).

The general rule is to specify first options and parameter arguments ("`--foo`" = option "foo" is on, "`--foo bar`" = parameter "foo" has value "bar"), and only then -- positional arguments, last of which can be optional and/or have multiple values.

That is why in other scripts we have this usage form:
```
script_name [<options>] [FILENAME...]
```

FILENAME here is the last positional argument, that is why it can be omitted (to enter the "stream mode"), or multiple files can be passed to the "file mode".

**About `shift`**

`shift [N]` (with default N=1) makes command line arguments to "move left" on N positions.
Say, you run script with arguments:
```
./my_script arg1 arg2 arg3
```
First your arguments are: "arg1", "arg2", "arg3". Now move them "to the left" on one position. As there is no position "to the left" of the "arg1", it disappears. To the left of "arg2" was "arg1" -- "arg2" goes to its position. Same for "arg3" and position of "arg2". Now the arguments you have are: "arg2", "arg3".

```
$ cat shift_test.sh 
#!/bin/bash

while [ $# -gt 0 ]; do
  echo "Arguments: $*"
  shift
done

$ ./shift_test.sh arg1 arg2 arg3
Arguments: arg1 arg2 arg3
Arguments: arg2 arg3
Arguments: arg3
```